### DISCUSSION TITLE
Uuid compile issue

### DISCUSSION THREAD

**Discussion by vatomicgreg** [diesel-rs/diesel#3388] [Q&A]
### Versions

- **Rust:**
- latest
- **Diesel:**
- 2.0.2
- **Database:**
- Postgres
- **Operating System**
OSX
### Feature Flags

- **diesel:**
uuid postgres
## Problem Description
Getting compile errors when adding a type UUID column

### What are you trying to accomplish?
Started with the simplest Diesel example and got it working.
Added a column to my Postgres database called uid with type UUID using this migration:
ALTER TABLE posts ADD COLUMN uid UUID

This autogenerated the following schema:

diesel::table! {
    posts (id) {
        id -> Int4,
        title -> Varchar,
        body -> Text,
        published -> Bool,
        uid -> Nullable<Uuid>,
    }
}

Then I modified my models.rs to look like this:

use diesel::{prelude::*, sql_types::Uuid};
use crate::schema::posts;

#[derive(Queryable)]
pub struct Post {
    pub id: i32,
    pub title: String,
    pub body: String,
    pub published: bool,
    pub uid: Uuid,
}

#[derive(Insertable)]
#[table_name = "posts"]
pub struct NewPost<'a> {
    pub title: &'a str,
    pub body: &'a str,
    pub uid: &'a Uuid,
}

Now it wont compile.  It gives the error on the last line:  pub uid: &'a Uuid

### What is the expected output?
To compile without errors

### What is the actual output?
the trait bound `diesel::sql_types::Uuid: diesel::Expression` is not satisfied
the following other types implement trait `diesel::Expression`:
  &'a T
  (T0, T1)
  (T0, T1, T2)
  (T0, T1, T2, T3)
  (T0, T1, T2, T3, T4)
  (T0, T1, T2, T3, T4, T5)
  (T0, T1, T2, T3, T4, T5, T6)
  (T0, T1, T2, T3, T4, T5, T6, T7)
and 140 others
required because of the requirements on the impl of `diesel::Expression` for `&'a diesel::sql_types::Uuid`
required because of the requirements on the impl of `AsExpression<diesel::sql_types::Nullable<diesel::sql_types::Uuid>>` for `&'a diesel::sql_types::Uuid`

### Steps to reproduce

My Cargo.toml has this for diesel:
diesel = { version = "2.0.2", features = ["uuid", "postgres"] }

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues)  and the [disussion forum](https://github.com/diesel-rs/diesel/discussions)  for similar possible closed issues.
 
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)
- [x] This issue can be reproduced without requiring a third party crate

**Comment by weiznich** [comment]
To cite the relevant part of the [documentation](https://docs.diesel.rs/master/diesel/index.html#serializing-and-deserializing):

> You can find all the types supported by Diesel in the sql_types module. These types are only used to represent a SQL type. You should never put them on your Queryable structs.

Other than that: Please do not use our issue tracker to ask for support, that's the wrong place. I've moved this one issue now to the discussion forum, which is the right place for such questions.

**Reply by vatomicgreg** [reply]
Thank you and I apologize for miscategorizing this as a bug rather than a discussion.  Very much a newbie.

So what type do I put in my Queryable structs and my Insertable structs to handle the Uuid columns?

(If this is not the place to ask such questions, should I move this discussion somewhere else?)

**Reply by weiznich** [reply]
This forum is the right place for questions, just don't use the issue tracker for that.

Your question as already answered in the [linked documentation](https://docs.diesel.rs/master/diesel/index.html#serializing-and-deserializing):

> To find all the Rust types which can be used with a given SQL type, see the documentation for that SQL type.

(That's literally the next sentence about that one linked above)
